tests/components/panel_iframe/test_init.py

```diff
@@ -1,100 +1,87 @@
 """The tests for the panel_iframe component."""
-import unittest
+import pytest
 
-from homeassistant import setup
 from homeassistant.components import frontend
+from homeassistant.setup import async_setup_component
 
-from tests.async_mock import patch
-from tests.common import get_test_home_assistant
 
+@pytest.mark.parametrize(
+    "config_to_try",
+    (
+        {"invalid space": {"url": "https://home-assistant.io"}},
+        {"router": {"url": "not-a-url"}},
+    ),
+)
+async def test_wrong_config(hass, config_to_try):
+    """Test setup with wrong configuration."""
+    assert not await async_setup_component(
+        hass, "panel_iframe", {"panel_iframe": config_to_try}
+    )
 
-class TestPanelIframe(unittest.TestCase):
-    """Test the panel_iframe component."""
 
-    def setUp(self):
-        """Set up things to be run when tests are started."""
-        self.hass = get_test_home_assistant()
-        self.addCleanup(self.hass.stop)
-
-    def test_wrong_config(self):
-        """Test setup with wrong configuration."""
-        to_try = [
-            {"invalid space": {"url": "https://home-assistant.io"}},
-            {"router": {"url": "not-a-url"}},
-        ]
-
-        for conf in to_try:
-            with patch(
-                "homeassistant.components.http.start_http_server_and_save_config"
-            ):
-                assert not setup.setup_component(
-                    self.hass, "panel_iframe", {"panel_iframe": conf}
-                )
-
-    def test_correct_config(self):
-        """Test correct config."""
-        with patch("homeassistant.components.http.start_http_server_and_save_config"):
-            assert setup.setup_component(
-                self.hass,
-                "panel_iframe",
-                {
-                    "panel_iframe": {
-                        "router": {
-                            "icon": "mdi:network-wireless",
-                            "title": "Router",
-                            "url": "http://192.168.1.1",
-                            "require_admin": True,
-                        },
-                        "weather": {
-                            "icon": "mdi:weather",
-                            "title": "Weather",
-                            "url": "https://www.wunderground.com/us/ca/san-diego",
-                            "require_admin": True,
-                        },
-                        "api": {"icon": "mdi:weather", "title": "Api", "url": "/api"},
-                        "ftp": {
-                            "icon": "mdi:weather",
-                            "title": "FTP",
-                            "url": "ftp://some/ftp",
-                        },
-                    }
+async def test_correct_config(hass):
+    """Test correct config."""
+    assert await async_setup_component(
```

Comment:
Codacy found an issue: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=6294088)